### PR TITLE
Deprecate `is_AbelianGroup`, `is_DualAbelianGroup`, `is_MatrixGroup`

### PR DESCRIPTION
**Pull request by mkoeppe**
### :memo: Checklist

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/groups/abelian_gps/abelian_group.py [hunk 1]
```diff
@@ -458,10 +458,16 @@ def is_AbelianGroup(x):
...
+        doctest:warning...
+        DeprecationWarning: the function is_AbelianGroup is deprecated;
+        use 'isinstance(..., AbelianGroup_class)' instead
+        See https://github.com/sagemath/sage/issues/37898 for details.
         True
         sage: is_AbelianGroup(AbelianGroup(7, [3]*7))
         True
     """
+    from sage.misc.superseded import deprecation
+    deprecation(37898, "the function is_AbelianGroup is deprecated; use 'isinstance(..., AbelianGroup_class)' instead")
```
Comment (mantepse) [inline src/sage/groups/abelian_gps/abelian_group.py:470]: shouldn't that be `use 'in Groups.Commutative'`, @fchapoton?

Comment (mantepse) [reply src/sage/groups/abelian_gps/abelian_group.py:470]: I just checked, yes, that's the pattern.  For example:
```
deprecation(35999, "the function is_CommutativeAlgebra is deprecated; use '... in Algebras(base_ring).Commutative()' instead")
```
or
```
deprecation(35253, "the function is_Algebra is deprecated; use '... in Algebras(base_ring)' instead")
```

Comment (mkoeppe) [reply src/sage/groups/abelian_gps/abelian_group.py:470]: I did use this pattern in these other deprecation messages when there was at least some indication that these `is_...` functions were used with this meaning.
But `is_AbelianGroup` has never been used with a meaning other than this specific `isinstance` test.

**Review by videlec — APPROVED** [review]

**Comment by mkoeppe** [comment]
Thanks!

**Pull request merged**